app.rb

```diff
@@ -89,4 +101,32 @@ class Application < Sinatra::Base
     erb(:peep_by_user)
   end
 
+  get '/login' do
+    erb(:login)
+  end
+
+  post '/login' do
+    email = params[:email_address]
+    password = params[:password]
+
+    user_repo = UserRepository.new
+    user = user_repo.find_by_email(email)
+
+    if user.password == password
+
+      session[:user_id] = user.id
+
+      return erb(:login_success)
+
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/logout' do
+    session.clear
+
+    redirect '/'
+  end
+
 end
```

Comment:
Your code is super clear and readable, which is great. Maybe remove a few of the empty spaces to make it even more succinct as a whole!